add: flatpost

ledif opened the pull request:
This adds [GloriousEggroll/flatpost](https://github.com/GloriousEggroll/flatpost), a desktop-environment-agnostic Flatpak browser and manager.

ledif committed:
Add flatpost

Co-authored-by: GloriousEggroll <gloriouseggroll@gmail.com>

ledif committed:
bump to 1.0.5

madonuko commented on anda/apps/flatpost/flatpost.spec:
```diff
@@ -0,0 +1,82 @@
+
+%changelog
+* Fri Jun 28 2024 Your Name <you@example.com> - 1.0-1
+- Initial package
```
You either add your name / email here, or remove the entire %changelog section (it's optional)

madonuko commented on anda/apps/flatpost/flatpost.spec:
```diff
@@ -0,0 +1,82 @@
+%license %{_datadir}/licenses/flatpost/LICENSE
+
+%clean
+rm -rf %{buildroot}
```
not needed

madonuko commented on anda/apps/flatpost/flatpost.spec:
```diff
@@ -0,0 +1,82 @@
+Requires: gtk4
+Requires: xdg-utils
+
+Provides: flatpost
```
this provides is not needed

madonuko commented on anda/apps/flatpost/flatpost.spec:
```diff
@@ -0,0 +1,82 @@
+# Set the default application for .rpm files
+xdg-mime default /usr/share/applications/com.flatpost.flatpostapp.desktop application/vnd.flatpak.ref
+xdg-mime default /usr/share/applications/com.flatpost.flatpostapp.desktop application/vnd.flatpak.repo
+update-mime-database /usr/share/mime
```
- you might need to update the mime database for %post, %postun and %posttrans.
- you probably should install something along the lines of `%_datadir/mime/packages/*.xml` instead of setting them manually
  - unfortunately upstream doesn't have this file, so you need to make it yourself
  - fortunately we have an example [here](https://github.com/terrapkg/packages/blob/0ceccc6f8ee18ad9df516653294f56621724503e/anda/games/osu-mime/osu-file-extensions.xml)
- `Requires(post): shared-mime-info` (and also postun and posttrans)

madonuko commented on anda/apps/flatpost/flatpost.spec:
```diff
@@ -0,0 +1,82 @@
+%global tag 1.0.5
+
+%global commit 4d3dc59ab28311ff90f640b88c5a156b349cc74b
+
+%global shortcommit %(c=%{commit}; echo ${c:0:7})
+
+%global build_timestamp %(date +"%Y%m%d")
+
+%global rel_build 1.%{build_timestamp}.%{shortcommit}%{?dist}
```
you don't need these; simply use `Release: 1%?dist`

madonuko reviewed (commented):
you also need `update.rhai`:
```ts
rpm.version(gh_rawfile("GloriousEggroll/flatpost", "main", "VERSION.txt"));
```
thanks for the submission

ledif committed:
partially address review comments

ledif committed:
add mime types

ledif commented:
Thanks for the thorough review comments. I believe I have addressed all of your points. Looking at the generated rpm, it seems to have the appropriate files.

![image](https://github.com/user-attachments/assets/af21c3f6-5bfd-4c1e-b2c3-46ce61934571)

madonuko commented on anda/apps/flatpost/flatpost-mime.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
+  <mime-type type="application/vnd.flatpak.ref">
```
Since you're using `application/vnd.flatpak.ref`, you need to edit the desktop file to include `MimeType=application/vnd.flatpak.ref`

ledif replied:
I added a patch to the .desktop file and opened a PR upstream https://github.com/GloriousEggroll/flatpost/pull/6. Here's the new .desktop file.
```
$ cat /tmp/usr/share/applications/com.flatpost.flatpostapp.desktop 
[Desktop Entry]
Name=Flatpost
Exec=python3 /usr/bin/flatpost
Icon=com.flatpost.flatpostapp
Type=Application
Categories=Utility;
MimeType=application/vnd.flatpak.ref;application/vnd.flatpak.repo;
```

madonuko commented on anda/apps/flatpost/flatpost.spec:
```diff
@@ -0,0 +1,74 @@
+Name:          flatpost
+Version:       %{tag}
+Release:       1%?dist
+License:       BSD 2-Clause
```
```suggestion
License:       BSD-2-Clause
```

madonuko commented:
nice theme damn

ledif committed:
Fix typo in License

Co-authored-by: madomado <madonuko@outlook.com>
Signed-off-by: Adam Fidel <ledif@users.noreply.github.com>

ledif committed:
add patch to add mime types

ledif commented:
> nice theme damn

This is [Blue95](https://blues.win/posts/blue95-topanga/) which uses the Chicago95 theme :)

raboneko commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|f40|[<img src="https://img.shields.io/github/pulls/detail/state/terrapkg/packages/4377">](https://github.com/terrapkg/packages/pull/4377)|
|✅|f41|[<img src="https://img.shields.io/github/pulls/detail/state/terrapkg/packages/4378">](https://github.com/terrapkg/packages/pull/4378)|
|✅|el10|[<img src="https://img.shields.io/github/pulls/detail/state/terrapkg/packages/4379">](https://github.com/terrapkg/packages/pull/4379)|
|✅|f42|[<img src="https://img.shields.io/github/pulls/detail/state/terrapkg/packages/4380">](https://github.com/terrapkg/packages/pull/4380)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport) and see the [Github Action logs](https://github.com/terrapkg/packages/actions/runs/14610401768) for details

<!--- Backport version: 9.3.0 -->
